src/main/java/edu/usf/sas/pal/muser/manager/MuserMusicDataAnalysisManager.java

```diff
@@ -62,27 +87,90 @@ private void analyzeAllUsersMusicData() {
         for (QueryDocumentSnapshot doc : userUiEventInfoById) {
-            MusicAnalysisModel mam = new MusicAnalysisModel(doc.toObject(MusicAnalysisModel.class), doc.getId(), userId);
-            musicAnalysisList.add(mam);
-            csvFileWriter.appendAllToCsV(musicAnalysisList);
-            musicAnalysisList.clear();
+            writeUserInfoToCsv(doc, userId);
+        }
+    }
+
+    private void processUserByIdForDeviceInfo(String userId) {
+        //get data of Device info  collection based on user id
+        List<QueryDocumentSnapshot> userDeviceInfoById = new ArrayList<>(firebaseReader.getDeviceInfoByUserId(userId));
+        for (QueryDocumentSnapshot doc : userDeviceInfoById) {
+            writeUserInfoToCsv(doc, userId);
         }
     }
+
+    private void processUserByDateRangeForPlayerEvent(String userId, long millisStart, long millisEnd) {
+        //get data of Player Events collection based on date range and user id
+        List<QueryDocumentSnapshot> userByDateRangeForPlayerEvent = new ArrayList<>(firebaseReader.getInfoByDateRangeForPlayerEventInfo(userId, millisStart, millisEnd));
+        for (QueryDocumentSnapshot doc : userByDateRangeForPlayerEvent) {
+            writeUserInfoToCsv(doc, userId);
+        }
+    }
+
+    private void processUserByIdDateRangeForUiEvent(String userId, long millisStart, long millisEnd) {
+        //get data of Ui Events info collection on date range and user id
+        List<QueryDocumentSnapshot> userByIdDateRangeForUiEvent = new ArrayList<>(firebaseReader.getInfoByDateRangeForUiEventInfo(userId, millisStart, millisEnd));
+        for (QueryDocumentSnapshot doc : userByIdDateRangeForUiEvent) {
+            writeUserInfoToCsv(doc, userId);
+        }
+    }
+
+    private void processUserByIdDateRangeForDeviceInfo(String userId, long millisStart, long millisEnd) {
+        //get data of device info collection on date range and user id
+        List<QueryDocumentSnapshot> userByIdDateRangeForUiEvent = new ArrayList<>(firebaseReader.getInfoByDateRangeForUiEventInfo(userId, millisStart, millisEnd));
+        for (QueryDocumentSnapshot doc : userByIdDateRangeForUiEvent) {
+            writeUserInfoToCsv(doc, userId);
+        }
+    }
+
+    private void writeUserInfoToCsv(QueryDocumentSnapshot doc, String userId) {
+        //Write the Music Analysis Model to CSV
+        MusicAnalysisModel mam = new MusicAnalysisModel(doc.toObject(MusicAnalysisModel.class), doc.getId(), userId);
+        musicAnalysisList.add(mam);
+        csvFileWriter.appendAllToCsv(musicAnalysisList);
+        musicAnalysisList.clear();
+    }
+
+    public long getStartDateInMillis() {
+        //below method converts start date to milliseconds
+        LocalDate startDate = LocalDate.parse(programOptions.getStartDate());
+        return startDate.atStartOfDay(ZoneId.of("America/New_York")).toInstant().toEpochMilli();
```

Comment:
Per related discussions in https://github.com/CUTR-at-USF/muser-firebase-export/issues/14, this should be in UTC time and not hard-coded to timezone for consistency.

Reply:
Please also make clear that UTC time is used in the command line help output and the README